### PR TITLE
 # 184581340 User can change his password(PATCH)

### PR DESCRIPTION
**Pull request by bencyubahiro77**
[ #184581340][the changed password will be saved in the database and replace the old one]

#### What does this PR do?
The PR provide ability to the user to update their password

#### Description of Task to be completed?
the user should be able to change his password if he/she wants too. he will provide the old password , then new password and the new password provided should be able to match the criteria of a valid password.

#### How should this be manually tested?
this can be tested using postman where you can
PATCH users/change-password
In the body insert
{
    "email": " ",
    "oldPassword": " ",
    "newPassword": " "
}

#### Any background context you want to provide?

n/a

#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/n/projects/2632830/stories/184581340

#### Screenshots (if appropriate)

Screenshot 2023-03-14 at 13 39 12

### CODE REVIEW

**Review by Martha-Iradukunda — APPROVED** [review]
LGTM

**Review by blackd44 — CHANGES_REQUESTED** [review]
Looks Good
but better change your pull request `name` and description add `PT story link`

**Review by Patienceineza — APPROVED** [review]
LGTM

File: src/controllers/users.controllers.js [hunk 1]
```diff
@@ -115,4 +115,39 @@ export default class Users {
...
     }
   }
+  static async changePassword(req, res) {
+    try {
+      const { email, oldPassword, newPassword } = req.body;
+
+      // check if the email exists in the database
+      const user = await db.users.findOne({ where: { email } });
+      if (!user) {
+        return res.status(404).json({ message: 'User not found' });
```
Comment (UmuhireJessie) [inline src/controllers/users.controllers.js:123]: Test this response

File: src/controllers/users.controllers.js [hunk 2]
```diff
@@ -115,4 +115,39 @@ export default class Users {
...
+      // check if the old password matches the one in the database
+      const isPasswordValid = bcrypt.compareSync(oldPassword, user.password);
+      if (!isPasswordValid) {
+        return res.status(401).json({ message: 'Incorrect old password' });
+      }
+      
+       // validate the new password
+    const passwordRegex = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]).{8,}$/;
+    if (!passwordRegex.test(newPassword)) {
+      return res.status(400).json({
```
Comment (UmuhireJessie) [inline src/controllers/users.controllers.js:135]: Test this too!

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
Hey @bencyubahiro77 

Please a screenshot of your test passing ,your swagger and also  update readme file

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Good job @bencyubahiro77 ,
One thing
1. Test all responses returned by your endpoint
Refer to this screenshot to see related lines of code in `user.controllers.js` file
Screenshot from 2023-03-14 15-14-13

**Review by edwinnambaje — CHANGES_REQUESTED** [review]
fix conflicts  and then you are good to go

**Review by blackd44 — APPROVED** [review]

**Pull request closed without merging**